src/go/rpk/pkg/cli/cmd/topic/consume_test.go

```diff
@@ -0,0 +1,262 @@
+package topic
+
+import (
+	"testing"
+	"time"
+)
+
+func TestParseFromToOffset(t *testing.T) {
+	for i, test := range []struct {
+		in string
+
+		start      int64
+		end        int64
+		rel        int64
+		atStart    bool
+		atEnd      bool
+		hasEnd     bool
+		currentEnd bool
+		expErr     bool
+	}{
+		{
+			expErr: true,
+		},
+
+		{
+			in:      "start",
+			atStart: true,
+		},
+		{
+			in:      "oldest",
+			atStart: true,
+		},
+
+		{
+			in:    "end",
+			atEnd: true,
+		},
+		{
+			in:    "newest",
+			atEnd: true,
+		},
+
+		{
+			in:         ":end",
+			atStart:    true,
+			currentEnd: true,
+		},
+
+		{
+			in:      "+3",
+			rel:     3,
+			atStart: true,
+		},
+		{
+			in:     "+a",
+			expErr: true,
+		},
+
+		{
+			in:    "-3",
+			rel:   -3,
+			atEnd: true,
+		},
+		{
+			in:     "-a",
+			expErr: true,
+		},
+
+		{
+			in:    "35",
+			start: 35,
+		},
+		{
+			in:    "35:",
+			start: 35,
+		},
+		{
+			in:      ":23",
+			end:     23,
+			atStart: true,
+			hasEnd:  true,
+		},
+
+		{
+			in:     "90:91",
+			start:  90,
+			end:    91,
+			hasEnd: true,
+		},
+		{
+			in:     "90-91",
+			start:  90,
+			end:    91,
+			hasEnd: true,
+		},
+		{
+			in:         "90:end",
+			start:      90,
+			currentEnd: true,
+		},
+		{
+			in:         "90-end",
+			start:      90,
+			currentEnd: true,
+		},
+
+		{
+			in:     "90:39",
+			expErr: true,
+		},
+		{
+			in:     "90:a",
+			expErr: true,
+		},
+
+		//
```

Comment:
Please fix the irregular inter-testcase spacing above and the unused comment 😅  It's making my brain go 😬

Reply:
The unused comment is a deliberate because it makes jumping around the test cases above easy in vim (the final test case being a separate empty line from the logic below), but I can remove it, as well as the squashed test cases above (which I wrote at the time being for separate blocks of code parseFromToOffset)

Reply:
Addressed!